src/averageAges.js

```diff
@@ -37,7 +41,20 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter(person => withChildren
+    ? isMother(people, person)
+    : person.sex === 'f'
+  );
+
+  return getAvagareAge(women);
+}
+
+function isMother(people, woman) {
+  return people.some(
+    child =>
+      child.mother === woman.name
+      && woman.sex === 'f'
+  );
```

Comment:
```suggestion
  return woman.sex === 'f' && people.some(
    child => child.mother === woman.name);
```
It is more optimal to check a person's sex before the array loop.